lib/seadsa/DsaLocal.cc

```diff
@@ -468,36 +479,55 @@ class IntraBlockBuilder : public InstVisitor<IntraBlockBuilder>,
   void visitAllocWrapperCall(CallSite &CS);
   void visitAllocationFnCall(CallSite &CS);
 
+  const llvm::StringRef m_seaFnModTag       = "sea_dsa_set_modified";
+  const llvm::StringRef m_seaFnReadTag      = "sea_dsa_set_read";
+  const llvm::StringRef m_seaFnPtrToIntTag  = "sea_dsa_set_ptrtoint";
+  const llvm::StringRef m_seaFnAliasTag     = "sea_dsa_alias";
+  const llvm::StringRef m_seaFnCollapseTag  = "sea_dsa_collapse";
+  const llvm::StringRef m_seaFnMkSeqTag     = "sea_dsa_mk_seq";
+
+  SeadsaFn getSeaDsaFn (const Function *fn) {
+    if (!fn)  return SeadsaFn::UNKNOWN;
+
+    if (fn->getName().equals(m_seaFnModTag))      return SeadsaFn::ALIAS;
+    if (fn->getName().equals(m_seaFnReadTag))     return SeadsaFn::READ;
+    if (fn->getName().equals(m_seaFnPtrToIntTag)) return SeadsaFn::PTR_TO_INT;
+    if (fn->getName().equals(m_seaFnAliasTag))    return SeadsaFn::ALIAS;
+    if (fn->getName().equals(m_seaFnCollapseTag)) return SeadsaFn::COLLAPSE;
+    if (fn->getName().equals(m_seaFnMkSeqTag))    return SeadsaFn::MAKE_SEQ;
+
+    return SeadsaFn::UNKNOWN;
+
+  }
+
   /// Returns true if \p F is a \p seadsa_ family of functions
   static bool isSeaDsaFn(const Function *fn) {
     if (!fn)
       return false;
     auto n = fn->getName();
-    return n.startswith("sea_dsa_") || n.startswith("seadsa_");
+    return n.startswith("sea_dsa_");
   }
 
-  static bool isSeaDsaAliasFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_alias") || name.equals("seadsa_alias");
-  }
-
-  static bool isSeaDsaCollapseFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_collapse") || name.equals("seadsa_collapse");
+  static bool isSeaDsaAttrbFn(const SeadsaFn &fn) {
+    switch (fn) {
+      case SeadsaFn::MODIFY:
+      case SeadsaFn::READ:
+      case SeadsaFn::PTR_TO_INT:
+        return true;
+      default:
+        return false;
+    }
   }
 
-  static bool isSeaDsaMkSequenceFn(const Function *fn) {
-    if (!fn)
-      return false;
-    auto name = fn->getName();
-    // due to name change from sea_dsa to seadsa, support both version
-    return name.equals("sea_dsa_mk_seq") || name.equals("seadsa_mk_seq");
+  auto getSeaDsaAttrbFunc(const Function *fn) {   
+    switch (getSeaDsaFn(fn)) {
```

Comment:
reminder: check that this works when `fn == nullptr` or change to `const Function &`